Gems/Atom/RPI/Code/Source/RPI.Public/RPIUtils.cpp

```diff
@@ -20,6 +20,203 @@ namespace AZ
+                case AZ::RHI::Format::R32_FLOAT:
+                {
+                    auto actualMem = reinterpret_cast<const float*>(mem);
+                    return actualMem[index];
+                }
+                default:
+                    AZ_Assert(false, "Unsupported pixel format");
+                    return 0.0f;
+                }
+            }
+
+            AZ::u32 RetrieveUintValue(const AZ::u8* mem, size_t index, AZ::RHI::Format format)
+            {
+                switch (format)
+                {
+                case AZ::RHI::Format::R8_UINT:
+                {
+                    return mem[index] / static_cast<AZ::u32>(std::numeric_limits<AZ::u8>::max());
+                }
+                case AZ::RHI::Format::R16_UINT:
+                {
+                    auto actualMem = reinterpret_cast<const AZ::u16*>(mem);
+                    return actualMem[index] / static_cast<AZ::u32>(std::numeric_limits<AZ::u16>::max());
+                }
+                case AZ::RHI::Format::R32_UINT:
+                {
+                    auto actualMem = reinterpret_cast<const AZ::u32*>(mem);
+                    return actualMem[index];
+                }
+                default:
+                    AZ_Assert(false, "Unsupported pixel format");
+                    return 0;
+                }
+            }
+
+            AZ::s32 RetrieveIntValue(const AZ::u8* mem, size_t index, AZ::RHI::Format format)
+            {
+                switch (format)
+                {
+                case AZ::RHI::Format::R8_SINT:
+                {
+                    return mem[index] / static_cast<AZ::s32>(std::numeric_limits<AZ::s8>::max());
+                }
+                case AZ::RHI::Format::R16_SINT:
+                {
+                    auto actualMem = reinterpret_cast<const AZ::s16*>(mem);
+                    return actualMem[index] / static_cast<AZ::s32>(std::numeric_limits<AZ::s16>::max());
+                }
+                case AZ::RHI::Format::R32_SINT:
+                {
+                    auto actualMem = reinterpret_cast<const AZ::s32*>(mem);
+                    return actualMem[index];
+                }
+                default:
+                    AZ_Assert(false, "Unsupported pixel format");
+                    return 0;
+                }
+            }
+
+            template<typename T>
+            T GetSubImagePixelValueInternal(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, uint32_t x, uint32_t y, uint32_t componentIndex, uint32_t mip, uint32_t slice)
+            {
+                AZStd::array<T, 1> values = { aznumeric_cast<T>(0) };
+
+                auto topLeft = AZStd::make_pair(x, y);
+                auto bottomRight = AZStd::make_pair(x + 1, y + 1);
+                AZStd::span<T> valueSpan(values.begin(), values.size());
+                GetSubImagePixelValues(imageAsset, topLeft, bottomRight, valueSpan, componentIndex, mip, slice);
```

Comment:
You can you can just pass the AZStd::array function directly into GetSubImagePixelValues function.
Also you can use Class Template Argument deduction for the array
```suggestion
                AZStd::array values{ aznumeric_cast<T>(0) };

                auto topLeft = AZStd::make_pair(x, y);
                auto bottomRight = AZStd::make_pair(x + 1, y + 1);
                GetSubImagePixelValues(imageAsset, topLeft, bottomRight, values, componentIndex, mip, slice);
```

Reply:
Will do, thanks!